packages/firestore/src/remote/online_state_tracker.ts

```diff
@@ -166,10 +170,18 @@ export class OnlineStateTracker {
     }
   }
 
-  private logClientOfflineWarningIfNecessary(): void {
+  private logClientOfflineWarningIfNecessary(details: string): void {
+    const message =
+      `Could not reach Cloud Firestore backend. ${details}\n` +
+      `This is purely an informational message and typically indicates ` +
```

Comment:
I would remove the first part of this sentence. Just saying "This typically indicates that your device ..." should make it obvious where the problem is.

Reply:
Good call, done.